src/CMakeLists.txt

```diff
@@ -482,21 +482,29 @@ if(WIN32)
   list(APPEND ceph_common_deps dlfcn_win32)
 endif()
 
-if(WITH_BLUESTORE_PMEM OR WITH_RBD_RWL)
+if(WITH_BLUESTORE_PMEM)
   find_package(ndctl 63 REQUIRED)
   find_package(daxctl 63 REQUIRED)
   if(WITH_SYSTEM_PMDK)
     set(pmdk_COMPONENTS)
-    if(WITH_BLUESTORE_PMEM)
-      list(APPEND pmdk_COMPONENTS pmem)
-    endif()
-    if(WITH_RBD_RWL)
-      list(APPEND pmdk_COMPONENTS pmemobj)
-    endif()
+    list(APPEND pmdk_COMPONENTS pmem)
+    find_package(COMPONENTS ${pmdk_COMPONENTS})
+  else()
+    include(Buildpmdk)
+    set(enable_ndctl ON)
+    build_pmdk(${enable_ndctl})
+  endif()
+endif()
+
+if(WITH_RBD_RWL)
+  if(WITH_SYSTEM_PMDK)
+    set(pmdk_COMPONENTS)
+    list(APPEND pmdk_COMPONENTS pmemobj)
     find_package(pmdk 1.8 REQUIRED COMPONENTS ${pmdk_COMPONENTS})
   else()
     include(Buildpmdk)
-    build_pmdk()
+    set(enable_ndctl OFF)
+    build_pmdk(${enable_ndctl})
```

Comment:
what if both `WITH_BLUESTORE_PMEM` and `WITH_RBD_RWL` are enabled?  i don't think we need to build pmdk twice in this case. could you restructure the existing code so it looks something like
```cmake
if(WITH_BLUESTORE_PMEM OR WITH_RBD_RWL)
  # ...
  if(WITH_SYSTEM_PMDK)
    if(WITH_BLUESTORE_PMEM)
      list(APPEND pmdk_COMPONENTS pmem)
    endif()
    if(WITH_RBD_RWL)
      list(APPEND pmdk_COMPONENTS pmemobj)
    endif()
    find_package(pmdk 1.8 REQUIRED
      COMPONENTS ${pmdk_COMPONENTS})
  else()
    include(Buildpmdk)
    set(enable_ndctl OFF)
    if(WITH_BLUESTORE_PMEM)
      set(enable_ndctl ON)
    endif()
    build_pmdk(${enable_ndctl})
  endif()
endif()
```

Reply:
Thanks your advice, revised and and add `find_package(ndctl 63 REQUIRED)` in `if(WITH_BLUESTORE_PMEM)`. please check.